runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -57,15 +58,28 @@ jvmtiGetStackTrace(jvmtiEnv* env,
 
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
+#if JAVA_SPEC_VERSION >= 19
+			if (NULL == targetThread) {
+				j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+				if (IS_VIRTUAL_THREAD(currentThread, threadObject)) {
+					j9object_t contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
+					rc = jvmtiInternalGetStackTrace(env, currentThread, NULL, contObject, start_depth, (UDATA) max_frame_count, frame_buffer, &rv_count);
+				} else {
+					Assert_JVMTI_unreachable();
```

Comment:
This assert is being repeated many times. We should move it into `getVMThread` where `targetThread` is set. This will also avoid the extra call to `IS_VIRTUAL_THREAD`.

Reply:
can you elaborate where the assertion should go?

Reply:
After `targetThread` is set in `getVMThread`:
https://github.com/eclipse-openj9/openj9/blob/7744616e477b18ce50a28150d431a136c0b3f04d/runtime/jvmti/jvmtiHelpers.c#L149

Reply:
Sorry I still don't get what should we assert there...

Reply:
In `getVMThread`, we are asserting that `targetThread` will only be NULL for a virtual thread.
```
Assert_JVMTI_true((NULL == targetThread) && isVirtualThread);
```
In `jvmtiGetStackTrace`, we can skip the `IS_VIRTUAL_THREAD` check and `Assert_JVMTI_unreachable` since `getVMThread` will guarantee `NULL == targetThread` only for a virtual thread.
```
#if JAVA_SPEC_VERSION >= 19
			if (NULL == targetThread) {
				j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
				j9object_t contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
				rc = jvmtiInternalGetStackTrace(env, currentThread, NULL, contObject, start_depth, (UDATA) max_frame_count, frame_buffer, &rv_count);
			} else
#endif /* JAVA_SPEC_VERSION >= 19 */
```
The above change will apply to all existing locations where `Assert_JVMTI_unreachable` was added.

Reply:
I think you probably mean `Assert_JVMTI_true((NULL != targetThread) || isVirtualThread);`?

Reply:
Yes. `targetThread` should never be NULL for a non-virtual thread. Your assertion is the negated version of ...
```
if ((NULL == targetThread) && !isVirtualThread) {
	Assert_JVMTI_unreachable();
}

OR

Assert_JVMTI_false((NULL == targetThread) && !isVirtualThread);
```